### DISCUSSION TITLE
onMount doesn't work for Atoms referenced in useAtomCallback

### DISCUSSION THREAD

**Discussion by zhmushan** [pmndrs/jotai#2053] [Bug report] [answered]
## Link to reproduction

https://codesandbox.io/s/boring-jackson-7d648z?file=/src/App.js

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
There doesn't seem docs clarifying the behavior, but `mount` in jotai means firstly subscribed. `useAtom` will subscribe to an atom, but `useAtomCallback` is a function to use an atom without subscription, so it's not mounted. We have similar discussions before and the current conclusion is this is by design to unmount atoms correctly.

**Comment by zhmushan** [comment]
@dai-shi Thanks your reply!

But this will result in the Atom used in useAtomCallback never being unmounted, and if it is a continuously running timer, it will execute indefinitely until it dies.

In my real-world application, multiple table components use the same filtering option, so I defined a filtering option atom. Different tables are switched through the Tab component, and the Atom that retrieves data is obtained in useAtomCallback. This causes the request for obtaining data from Table A to continue to be sent when I switch from Table A to Table B after updating the filtering option atom. I don't know what the best practice of jotai is for dealing with this scenario. (Because I want to manually control the Loading status of the page and I must delay setting the data, I am using useAtomCallback here.)

**Reply by dai-shi** [reply]
I don't know what's the exact problem with your scenario with `tableAAtom`, `tableBAtom` and common `filteringAtom`. Would you be able to create a very very small repro with https://csb.jotai.org?

**Reply by zhmushan** [reply]
@dai-shi 
https://codesandbox.io/s/musing-raman-lcmctg?file=/src/App.tsx

**Reply by dai-shi** [reply]
Can you describe what's your expected behavior?

**Reply by zhmushan** [reply]
@dai-shi  First, when clicking "Change Filter", we noticed that only one request was made to /tableA. Then we switched to Table B and clicked "Change Filter" again, which resulted in two requests being made to /tableA and /tableB. I want it to only make one request to /tableB.

**Reply by dai-shi** [reply, marked as answer]
I see.

> I don't know what the best practice of jotai is for dealing with this scenario. (Because I want to manually control the Loading status of the page

If you want such a control, it means it's not declarative, but imperative. Whatever it means, it's something like this in Jotai best practice:
```js
export const tableAFilterAtom = atom({});
export const tableADataAtom = atom({});
export const setTableADataAtom = atom(null, async (get, set) => {
  const filter = get(filterAtom);
  const tableAFilter = get(tableAFilterAtom);

  try {
    const data = await fetch("/tableA", {
      method: "POST",
      body: JSON.stringify({ ...filter, ...tableAFilter })
    });
  } catch {}

  set(tableADataAtom, { type: "A" });
});
```

**Reply by zhmushan** [reply]
Thank you so much for your patient and helpful answers. Your assistance has been greatly appreciated. Thank you again!

**Reply by Qquanwei** [reply]
In the scenario of codesandbox, I think sending two network requests is not expected. Is this a known bug of jotai?
